batch/batch/front_end/front_end.py

```diff
@@ -1125,10 +1138,29 @@ async def cancel_batch(request, userdata, batch_id):  # pylint: disable=unused-a
     return web.Response()
 
 
-@routes.patch('/api/v1alpha/batches/{batch_id}/close')
+async def _commit_jobs(app, db, batch_id, user, now=None):
+    try:
+        if now is None:
+            now = time_msecs()
+        await check_call_procedure(db, 'CALL commit_staged_jobs(%s, %s);', (batch_id, now))
+    except CallError as e:
+        if e.rv['rc'] == 1:
+            raise web.HTTPBadRequest(reason=f'batch {batch_id} has already been closed')
+        raise
+
+    async with client_session() as session:
+        await request_retry_transient_errors(
+            session,
+            'PATCH',
+            deploy_config.url('batch-driver', f'/api/v1alpha/batches/{user}/{batch_id}/commit'),
+            headers=app['batch_headers'],
+        )
+
+
+@routes.patch('/api/v1alpha/batches/{batch_id}/commit')
```

Comment:
I don't understand the motivation for this. In my head, the conversation between client and server would go like this:
```
> create batch
< OK, batch 10 created
> create jobs ...
< OK
```
Any job with no parents is eligible for scheduling now.
```
> create jobs ...
< OK
```
Any job whose parents are completed is eligible for scheduling.
```
> batch 10 status
< NOT DONE
... exponential back off of status requests
> batch 10 status
< DONE
```

In this conversation there's only three endpoints: 
1. create batch`POST /api/v1alpha/batches/create`
2. create jobs `POST /api/v1alpha/batches/{batch_id}/jobs/create`
3. get batch `GET /api/v1alpha/batches/{batch_id}`

I think we still need batch cancel and batch delete.

I think what I'm looking for is:
1. What use-cases does commit support?
2. Why does the user need to call close? Put another way: what goes wrong if a batch is "closed" 24 hours later?

I think there are four batch states:
1. accepting-new-jobs
2. not-accepting-new-jobs
3. cancelled 
4. deleted

I think the notion of "complete" for a batch doesn't quite make sense any more. In `hailtop.batch` we can still define "complete" as: "BatchBuilder.run was called and n_jobs == n_completed_jobs"